packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -122,49 +171,81 @@ export const prepareStatusManager = zone => {
       },
 
       /**
-       * Find an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`
+       * @param {CctpTxEvidence} evidence
+       */
+      isSeen(evidence) {
+        const seenKey = seenTxKeyOf(evidence);
+        return seenTxs.has(seenKey);
+      },
+
+      /**
+       * Remove and return an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`.
        *
        * @param {NobleAddress} address
        * @param {bigint} amount
-       * @returns {boolean}
+       * @returns {Pick<PendingTx, 'status' | 'txHash'> | undefined} undefined if nothing is settleable
        */
-      hasPendingSettlement(address, amount) {
+      dequeueStatus(address, amount) {
         const key = makePendingTxKey(address, amount);
+        if (!pendingTxs.has(key)) return undefined;
         const pending = pendingTxs.get(key);
-        return !!pending.length;
+        if (!pending.length) return undefined;
```

Comment:
```suggestion
```
We might be able to remove this - it seems unreachable with the `pendingTxs.delete` logic in `dequeueStatus`. _(Also the only line popping up on `test:c8`)_

Reply:
it's also taken care of by the `dequeueIdx < 0` check after `find`. it's at best an optimization.